### DISCUSSION TITLE
"fp16_run": true in the config improves quality?

### DISCUSSION THREAD

**Discussion by Lordmau5** [voicepaw/so-vits-svc-fork#288] [General]
Okay so I've looked around the config a bit, seeing as I wasn't too happy with the results of a model that, while having a very small dataset, doesn't sound too great even after around 20k iterations. (A similar voice with even less data, only around 30s of voice clips, sounded better at 4k already)

So I noticed this option in the config file in `configs\44k\config.json` after running `svc pre-config`
```json
{
  "train": {
    ...
    "fp16_run": false,
    ...
  }
}
```

Changing that to `true` made the model already sound much better at even 200 and 400 iterations.

On another note, it didn't improve the ~20k model when I changed that setting. It had to be done on a 0 checkpoint / fresh model.

Is there anyone else that would want to give this a try and report back?

- `svc pre-config`
- Change the value in `configs\44k\config.json` to `true`
- Train your model from scratch as usual

**Comment by sbersier** [comment]
Indeed, setting fp16_run = true seems to give better results. 
I compared results at 800 iterations for both options, as well as results for 20'000 iterations for both options.
Now, 20'000 is clearly not enough (whatever fp16_run is set to).

I  have a voice that I trained for 105600 iterations with the default fp16_run = false

I will retrain the whole thing with fp16_run = true,  Just to see...

**Comment by sbersier** [comment]
So, I trained a voice for 105600 iterations with a) fp16_run set to true and b) fp16_run set to false  (the default)

https://user-images.githubusercontent.com/34165937/231388261-e5e44556-514e-43b4-b289-662a723ff96c.mp4

In my opinion, the default (fp16_run = false) seems to be cleaner in the long run. 
As you said, at the begining of the training, fp16_run = true (non default) seems to be a bit better but the advantage seems to wear off as the training progresses.

Note: The generation was done with: 
`svc infer -n 1 -a -fm crepe -m SP_A_fp16_false/logs/44k/G_105600.pth -c SP_A_fp16_false/logs/44k/config.json input.wav -o output_fp16_false_105600.wav`
